velox/expression/CastExpr.cpp

```diff
@@ -247,6 +247,30 @@ VectorPtr CastExpr::castToDate(
   }
 }
 
+VectorPtr CastExpr::castIntToTimestamp(
+    const SelectivityVector& rows,
+    const BaseVector& input,
+    exec::EvalCtx& context) {
+  VectorPtr castResult;
+  context.ensureWritable(rows, TIMESTAMP(), castResult);
+  (*castResult).clearNulls(rows);
+  auto* resultFlatVector = castResult->as<FlatVector<Timestamp>>();
+  auto* inputFlatVector = input.as<SimpleVector<int64_t>>();
+  const auto& queryConfig = context.execCtx()->queryCtx()->queryConfig();
+  const auto sessionTzName = queryConfig.sessionTimezone();
+  const auto* timeZone =
+      (queryConfig.adjustTimestampToTimezone() && !sessionTzName.empty())
+      ? date::locate_zone(sessionTzName)
+      : nullptr;
+  applyToSelectedNoThrowLocal(context, rows, castResult, [&](int row) {
+    auto timestamp = Timestamp::fromMillis(
+        inputFlatVector->valueAt(row) * kMillisInSecond);
```

Comment:
Would you align the implementation with longToTimestamp method in Spark? Overflow needs to be handled.
https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L692-L693